ospfd/ospf_auth.c

```diff
@@ -222,25 +229,29 @@ static int ospf_auth_check_md5_digest(struct ospf_interface *oi,
 static int ospf_auth_make_md5_digest(struct ospf_interface *oi,
 				     struct ospf_packet *op, struct key *key)
 {
-	void *ibuf;
-	struct ospf_header *ospfh;
+	void *ibuf = STREAM_DATA(op->s);
+	struct ospf_header *ospfh = (struct ospf_header *)ibuf;
 	unsigned char digest[OSPF_AUTH_MD5_SIZE];
-	uint16_t length;
+	uint16_t length = ntohs(ospfh->length);
 #ifdef CRYPTO_OPENSSL
 	EVP_MD_CTX *ctx;
 #elif CRYPTO_INTERNAL
 	MD5_CTX ctx;
 #endif
 	char auth_key[OSPF_AUTH_MD5_SIZE + 1];
 
+	if ((length < (sizeof(struct ospf_header))) || (length > op->length)) { /* for coverity's sake */
+		flog_warn(EC_OSPF_AUTH,
+			  "interface %s: Invalid packet length, Router-ID: %pI4",
```

Comment:
Same as the previous comment about the length. Also, please use different text for warning message from the previous one, to distinguish which is which. Otherwise, we end up guessing which part was hit.

Reply:
Added packet length and `__func__` to the message for separation of warnings and giving out more information.